components/gitpod-cli/cmd/ports-list.go

```diff
@@ -48,21 +48,40 @@ var listPortsCmd = &cobra.Command{
 		table.SetCenterSeparator("|")
 
 		for _, port := range ports {
-			status := "not served"
+			status := ""
 			statusColor := tablewriter.FgHiBlackColor
-			if port.Exposed == nil && port.Tunneled == nil {
+			accessible := port.Exposed != nil || port.Tunneled != nil
+
+			exposedUrl := ""
+			if port.Exposed != nil {
+				exposedUrl = port.Exposed.Url
+			}
+
+			if !port.Served {
+				status = "not served"
+			} else if !accessible {
 				if port.AutoExposure == supervisor.PortAutoExposure_failed {
 					status = "failed to expose"
 					statusColor = tablewriter.FgRedColor
 				} else {
 					status = "detecting..."
 					statusColor = tablewriter.FgYellowColor
 				}
-			} else if port.Served {
-				status = "open (" + port.Exposed.Visibility.String() + ")"
-				if port.Exposed.Visibility == supervisor.PortVisibility_public {
+			} else {
```

Comment:
_Suggestion_: How about saving some processing by reducing the checks?
Currently, it will run between 4 and 8 checks. If you change it, it will run from 2 to 4 checks:

```
			} else if port.Tunneled != nil {
				if port.Tunneled.Visibility == supervisor.TunnelVisiblity(supervisor.TunnelVisiblity_value["network"]) {
					status = "open on all interfaces"
					statusColor = tablewriter.FgHiGreenColor
				} else if port.Tunneled.Visibility == supervisor.TunnelVisiblity(supervisor.TunnelVisiblity_value["host"]) {
					status = "open on localhost"
					statusColor = tablewriter.FgHiGreenColor
				}
			} else if port.Exposed != nil {
				if port.Exposed.Visibility == supervisor.PortVisibility_public {
					status = "open (public)"
					statusColor = tablewriter.FgHiGreenColor
				} else if port.Exposed.Visibility == supervisor.PortVisibility_private {
					status = "open (private)"
					statusColor = tablewriter.FgHiCyanColor
				}
			}
```

Reply:
thanks for the feedback, @felladrin :) I've integrated the suggestion